build.gradle

```diff
@@ -87,3 +96,99 @@ jar.dependsOn generateClientPropertiesFile
+            "**/lightcouch/CouchDbException.java",
+            "**/client/**")
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+artifacts {
+    archives sourcesJar, javadocJar
+}
+
+//load signing and repository parameters from system properties
+['signing.keyId', 'signing.password', 'signing.secretKeyRingFile', 'ossrhUsername', 'ossrhPassword']
+        .each { propName ->
+    //set a property with the given name if the system property is set
+    //if the system property is not set then set the property to null if it isn't a signing one
+    if (System.properties.propName != null || !propName.startsWith("signing")) {
+        ext.(propName.toString()) = System.properties.propName
+    }
+}
+
+signing {
+    //only apply signing when it is a release and is being published
+    required {
+        isReleaseVersion && gradle.taskGraph.hasTask("uploadArchives")
+    }
+    //when signing, sign the archives
+    sign configurations.archives
+}
+
+uploadArchives {
+    repositories {
+        mavenDeployer {
+
+            //when publishing sign the pom
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: ossrhUsername, password: ossrhPassword)
+            }
+
+            //augment the pom with additional information
+            pom.project {
+                packaging 'jar'
+                name project.clientName
+                description 'official Cloudant Client for Java'
+                inceptionYear '2014'
+                url 'http://www.cloudant.com'
+                licenses {
+                    license {
+                        name 'The Apache Software License, Version 2.0'
+                        url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                        distribution 'repo'
+                    }
+                }
+                scm {
+                    connection 'scm:git:git://github.com/cloudant/java-cloudant.git'
+                    developerConnection 'scm:git:git@github.com/cloudant/java-cloudant.git'
+                    url 'https://java-cloudant@github.com/cloudant/java-cloudant.git'
+                }
+                properties {
+                    'project.build.sourceEncoding' 'UTF-8'
+                }
+                developers {
+                    developer {
+                        id 'IBMCloudant'
+                        name 'IBMCloudant'
+                        email 'github@cloudant.com'
+                        url 'https://github.com/cloudant/java-cloudant'
+                    }
```

Comment:
From looking at [cloudant toolkit]( https://repo1.maven.org/maven2/com/ibm/mobile/services/cloudant-toolkit-local/1.0.0/cloudant-toolkit-local-1.0.0.pom) pom file, MF used Mobile First SDK as the developer, so I think we should use IBM Cloudant, but have IBM as the organisation name with the links pointing to cloudant.com. Although I guess we could have the organisation Url pointing to IBM.com and the developer URL pointing to cloudant.com.